### PR TITLE
feat(influxdbexporter): limit size of payload

### PR DESCRIPTION
**Pull request by jacobmarble**
**Description:** 

InfluxDB often responds to exporter write requests with `413 Request Entity Too Large` because the line protocol payload is too large. (The otelcol batch processor can be used to limit batch size, but not in terms of bytes out put to InfluxDB.)

To fix this, the influxdbexporter should self-limit based on both bytes and lines of line protocol, defaulting to [the documented suggested limits](https://docs.influxdata.com/influxdb/cloud-serverless/write-data/best-practices/optimize-writes/#batch-writes) of 10,000 lines and 10MB. The service hard limit is 50MB [as documented here](https://docs.influxdata.com/influxdb/cloud-serverless/api/#operation/PostWrite).

Also tracked here: https://github.com/influxdata/influxdb-observability/issues/262

**Link to tracking Issue:** https://github.com/influxdata/influxdb-observability/issues/262

**Testing:** Added unit test

**Documentation:** Updated README and default config

### CODE REVIEW

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -438,7 +438,7 @@ require (
 	github.com/influxdata/go-syslog/v3 v3.0.1-0.20210608084020-ac565dc76ba6 // indirect
 	github.com/influxdata/influxdb-observability/common v0.5.2 // indirect
 	github.com/influxdata/influxdb-observability/influx2otel v0.5.2 // indirect
-	github.com/influxdata/influxdb-observability/otel2influx v0.5.2 // indirect
+	github.com/influxdata/influxdb-observability/otel2influx v0.0.0-20230705233630-3cfabd1d00b9 // indirect
```
Comment (codeboten) [inline cmd/otelcontribcol/go.mod:441]: should this not be `v0.5.2-1...` or something like it?

**Review by codeboten — COMMENTED** [review]
Restarted the CI, looks like the failed test already has an open issue for it

File: cmd/otelcontribcol/go.mod (hunk 1 above)
Comment (jacobmarble) [reply cmd/otelcontribcol/go.mod]: Fixed. (You forced me to finally learn how that works -- not complicated! https://go.dev/ref/mod#pseudo-versions )

**Review by codeboten — COMMENTED** [review]
Needs a `make gotidy`

**Comment by jacobmarble** [comment]
> Needs a `make gotidy`

Done.

File: cmd/configschema/go.mod [hunk 2]
```diff
@@ -429,7 +429,7 @@ require (
 	github.com/influxdata/go-syslog/v3 v3.0.1-0.20210608084020-ac565dc76ba6 // indirect
 	github.com/influxdata/influxdb-observability/common v0.5.2 // indirect
 	github.com/influxdata/influxdb-observability/influx2otel v0.5.2 // indirect
-	github.com/influxdata/influxdb-observability/otel2influx v0.5.2 // indirect
+	github.com/influxdata/influxdb-observability/otel2influx v0.5.3-0.20230705233630-3cfabd1d00b9 // indirect
```
Comment (mx-psi) [inline cmd/configschema/go.mod:432]: Not strictly required, but any chance we can use a stable release instead?

Comment (jacobmarble) [reply cmd/configschema/go.mod:432]: (This is true because the upstream change breaks the API, so this is an exceptional situation.)

**Review by mx-psi — APPROVED** [review]

**Comment by jacobmarble** [comment]
> Restarted the CI, looks like the failed test already has an open issue for it

In case that CI issue has been fixed, I rebased to trigger CI again.

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**